src/routes/conversation/[id]/+server.ts

```diff
@@ -232,13 +233,25 @@ export async function POST({ request, locals, params, getClientAddress }) {
 				}
 			);
 
-			let webSearchResults: WebSearch | undefined;
+			let webSearchResults: RagContextWebSearch | undefined;
 
 			if (webSearch) {
-				webSearchResults = await runWebSearch(conv, newPrompt, update);
+				webSearchResults = (await RAGs["webSearch"].retrieveRagContext(
+					conv,
+					newPrompt,
+					update
+				)) as RagContextWebSearch;
 			}
 
-			messages[messages.length - 1].webSearch = webSearchResults;
+			messages[messages.length - 1].ragContext = webSearchResults;
+
+			let pdfSearchResults: RagContext | undefined;
+			const pdfSearch = await collections.files.findOne({ filename: `${convId.toString()}-pdf` });
+			if (pdfSearch) {
+				pdfSearchResults = await RAGs["pdfChat"].retrieveRagContext(conv, newPrompt, update);
+			}
+
+			messages[messages.length - 1].ragContext = pdfSearchResults;
```

Comment:
This causes a bug with web search.
messages[messages.length - 1].ragContext = webSearchResults is being overridden with  undefined if no pdf document has been uploaded by the user.

Reply:
Submitted a PR to fix the issue #745